clang/test/CheckedC/ast-dump-where-clauses.c

```diff
@@ -0,0 +1,120 @@
+  // CHECK-AST-NEXT: IntegerLiteral {{.*}} 3
+  // CHECK-AST-NEXT: WhereClause
+  // CHECK-AST-NEXT:   RelationalFact
+  // CHECK-AST-NEXT:     BinaryOperator {{.*}} '>='
+  // CHECK-AST-NEXT:       ImplicitCastExpr {{.*}}
+  // CHECK-AST-NEXT:         DeclRefExpr {{.*}} 'x'
+  // CHECK-AST-NEXT:       IntegerLiteral {{.*}} 'int' 31
+  // CHECK-AST-NEXT:   RelationalFact
+  // CHECK-AST-NEXT:     BinaryOperator {{.*}} '<'
+  // CHECK-AST-NEXT:       ImplicitCastExpr {{.*}}
+  // CHECK-AST-NEXT:         DeclRefExpr {{.*}} 'x'
+  // CHECK-AST-NEXT:       IntegerLiteral {{.*}} 'int' 32
+
+  int x4 = 4 _Where x >= 31 _And p : count(x) _And x <= 34;
+  // CHECK-AST: VarDecl {{.*}} x4
+  // CHECK-AST-NEXT: IntegerLiteral {{.*}} 4
+  // CHECK-AST-NEXT: WhereClause
+  // CHECK-AST-NEXT:   RelationalFact
+  // CHECK-AST-NEXT:     BinaryOperator {{.*}} '>='
+  // CHECK-AST-NEXT:       ImplicitCastExpr {{.*}}
+  // CHECK-AST-NEXT:         DeclRefExpr {{.*}} 'x'
+  // CHECK-AST-NEXT:       IntegerLiteral {{.*}} 'int' 31
+  // CHECK-AST-NEXT:   BoundsFact: p
+  // CHECK-AST-NEXT:     CountBoundsExpr {{.*}}
+  // CHECK-AST-NEXT:       ImplicitCastExpr {{.*}}
+  // CHECK-AST-NEXT:         DeclRefExpr {{.*}} 'x'
+  // CHECK-AST-NEXT:   RelationalFact
+  // CHECK-AST-NEXT:     BinaryOperator {{.*}} '<='
+  // CHECK-AST-NEXT:       ImplicitCastExpr {{.*}}
+  // CHECK-AST-NEXT:         DeclRefExpr {{.*}} 'x'
+  // CHECK-AST-NEXT:       IntegerLiteral {{.*}} 'int' 34
+
+  int x5 = 2;
+
+  _Where x > 6 _And x < 62;    
+  // CHECK-AST: NullStmt
+  // CHECK-AST-NEXT: WhereClause
+  // CHECK-AST-NEXT:   RelationalFact
+  // CHECK-AST-NEXT:     BinaryOperator {{.*}} '>'
+  // CHECK-AST-NEXT:       ImplicitCastExpr {{.*}}
+  // CHECK-AST-NEXT:         DeclRefExpr {{.*}} 'x'
+  // CHECK-AST-NEXT:       IntegerLiteral {{.*}} 'int' 6
+  // CHECK-AST-NEXT:   RelationalFact
+  // CHECK-AST-NEXT:     BinaryOperator {{.*}} '<'
+  // CHECK-AST-NEXT:       ImplicitCastExpr {{.*}}
+  // CHECK-AST-NEXT:         DeclRefExpr {{.*}} 'x'
+  // CHECK-AST-NEXT:       IntegerLiteral {{.*}} 'int' 62
+
+  x5 = 42 _Where x5 > 41;
+  // CHECK-AST: BinaryOperator {{.*}} '='
+  // CHECK-AST-NEXT: DeclRefExpr {{.*}} 'x5' 'int'
+  // CHECK-AST-NEXT: IntegerLiteral {{.*}} 'int' 42
+  // CHECK-AST-NEXT: WhereClause
+  // CHECK-AST-NEXT:   RelationalFact
+  // CHECK-AST-NEXT:     BinaryOperator {{.*}} '>'
+  // CHECK-AST-NEXT:       ImplicitCastExpr {{.*}}
+  // CHECK-AST-NEXT:         DeclRefExpr {{.*}} 'x5'
+  // CHECK-AST-NEXT:       IntegerLiteral {{.*}} 'int' 41
+
+  ++x5 _Where x5 > 41;
+  // CHECK-AST: UnaryOperator {{.*}} '++'
+  // CHECK-AST-NEXT: DeclRefExpr {{.*}} 'x5' 'int'
+  // CHECK-AST-NEXT: WhereClause
+  // CHECK-AST-NEXT:   RelationalFact
+  // CHECK-AST-NEXT:     BinaryOperator {{.*}} '>'
+  // CHECK-AST-NEXT:       ImplicitCastExpr {{.*}}
+  // CHECK-AST-NEXT:         DeclRefExpr {{.*}} 'x5'
+  // CHECK-AST-NEXT:       IntegerLiteral {{.*}} 'int' 41
+}
```

Comment:
Nit: please add a newline at the end of the file.

Reply:
Got it.